modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TreeTableRowSkinTest.java

```diff
@@ -320,6 +323,20 @@ public void invisibleColumnsShouldRemoveCorrespondingCellsInRow() {
         invisibleColumnsShouldRemoveCorrespondingCellsInRowImpl();
     }
 
+    /** TreeTableView.refresh() must release all discarded cells JDK-8307538 */
+    @Test
+    public void cellsMustBeCollectableAfterRefresh() {
+        IndexedCell<?> row = VirtualFlowTestUtils.getCell(treeTableView, 0);
+        assertNotNull(row);
+        WeakReference<Object> ref = new WeakReference<>(row);
+        row = null;
+
+        treeTableView.refresh();
+        Toolkit.getToolkit().firePulse();
```

Comment:
I think cells should also be collectable in other situations, like replacing the row factory.  Is it worth adding a test for that?

Reply:
this test reflects the exact failure scenario described in the ticket.

Reply:
If it's easy to add another test for additional missing cases that would provide better coverage for your fix, it seems worth doing.

Reply:
added one more test case.